HyperPlay-Gaming/hyperplay-desktop-client
# [TECH] Auto-Update improvements

### Changes

- Check for updates every hours instead of 12h.
- Download in the background and show only a single Dialog asking to restart.
- If Yes, apply update then restart.
- If No, apply the update after closing the app and reopening.
- Fix showing two dialogs on every update.

## How to Test

To test auto update on Windows locally make sure you added the option `"verifyUpdateCodeSignature": false`
under `build.win` in package.json  as bellow and change the app version to an old number, eg.: 0.17.1 and build it with `yarn dist:win` from the terminal.

Install the build version from the dist folder and check if it works.


package.json snippet:
```
    "win": {
      "artifactName": "${productName}-${version}-Setup-${arch}.${ext}",
      "icon": "build/win_icon.ico",
      "asarUnpack": [
        "build/bin/win32/legendary.exe",
        "build/bin/win32/gogdl.exe"
      ],
      "files": [
        "build/bin/win32/*"
      ],
      "verifyUpdateCodeSignature": false
    },
```

AND

` "version": "0.17.1"`


---

Use the following Checklist if you have changed something on the Backend or Frontend:
